File: 392-is-subsequence/solution.cpp
#include <string>

class Solution
{
public:
  bool isSubsequence(std::string s, std::string t)
  {
    if (s.size() > t.size())
    {
      return false;
    }

    size_t s_counter = 0;
    for (const char c : t)
    {
      if (s_counter >= s.size())
      {
        break;
      }

      if (c == s[s_counter])
      {
        ++s_counter;
      }
    }

    return s_counter == s.size();
  }
};

File: 392-is-subsequence/solution.ts
export function isSubsequence(s: string, t: string): boolean {
  if (s.length > t.length) {
    return false;
  }

  let sCounter = 0;
  for (const c of t) {
    if (sCounter >= s.length) {
      break;
    }

    if (c === s[sCounter]) {
      ++sCounter;
    }
  }

  return sCounter === s.length;
}
